docs/blueprints/EP029.rst

```diff
@@ -0,0 +1,103 @@
+
+ - `primary_path` (list of Links): list of Links describing the preferred route/path to be used by the EVC
+ - `backup_path` (list of Links): list of Links describing the backup/alternative route/path to be used by the EVC when the `primary_path` has any failure or unavailability condition (see section Protection Terminology)
+ - `dynamic_backup_path` (boolean): whether or not the path should be dynamic (internally obtained from pathfinder). This parameter provides three use cases: a) totally dynamic EVC -- meaning all the paths will be dynamic; b) only the backup path is dynamic -- when the user requests a primary_path and dynamic_backup_path; c) second-level backup path is dynamic -- when the user requests a primary_path, a backup_path, and dynamic_backup_path.
+
+In all scenarios above, when the Kytos SDN controller observes a link failure, the following procedure is applied:
+ 1. All EVCs are checked to verify if it is affected by the link failure;
+ 2. When a particular EVC is affected, the current_path is unconfigured (FlowMod Removal requests are sent to each switch in the current path);
+ 3. A dynamic or static path is determined, either from pathfinder (for dynamic EVCs) or user-provided parameters (static EVCs), along with the hop-by-hop service provider VLAN tag;
+ 4. The path from step 3 is configured (FLowMod Insert requests are sent to each switch)
+ 5. UNIs are reconfigured (FlowMod Insert requests are sent to UNI_A and UNI_Z's switches).
+
+Those steps are applied for each EVC, impacting the overall restoration time. Therefore, the restoration time of a particular EVC has its overhead (steps 2 to 5) plus the restoration time of all previous EVCs.
+
+
+Protection Terminology
+======================
+
+This section provides some protection terminology adopted by this document.
+
+Failure Types and Unavailability
+--------------------------------
+
+Failures may occur in network nodes or on the links between nodes. The following list describes the failures considered by this document:
+
+ - Fail condition: Fail condition is a status of a resource in which it is unable to transfer traffic (e.g., Loss of Signal, port down)
+ - Degrade condition: Degrade Condition is a resource status in which traffic transfer might be continuing, but specific measured errors (e.g., Bit Error Rate, etc.) has reached a pre-determined threshold.
+ - Node Failure: A Node Failure is an event that occurs when a node is unable to transfer traffic between the links that terminate at it. Note that a note may have a failure in the control plane but not in the data plane, meaning: the node can forward traffic between links, but the node does not respond to control plane and management plane requests.
+ - Unavailability: unavailable resources are the ones that were administratively configured not to be used (e.g., due to a maintenance window, traffic engineering, etc.)
+
+Path Disjointness
```

Comment:
Mapped issue kytos-ng/pathfinder#23 to migrate this requirement for pathfinder later on.